tool/tctl/common/collection.go

```diff
@@ -802,6 +802,13 @@ type windowsDesktopAndServiceCollection struct {
 	verbose  bool
 }
 
+func (c *windowsDesktopAndServiceCollection) windowsDesktops() (r []types.WindowsDesktop) {
```

Comment:
I wonder if we even need the `windowsDesktopAndServiceCollection`. I don't recall the details behind why that was added, but they are different resource types and I can't think of any reason we'd ever want to see them mixed.

Reply:
The `windows_desktop_service` is being fetched in order to show the `Host` & `Version` when `--format text`:

https://github.com/gravitational/teleport/blob/a2b5a37f94f9e0c134a4228c53d4044a04548169/tool/tctl/common/collection.go#L819-L824

For each of the `ls` command, we seem to have info about the following resources:
- `tctl nodes ls`: `node`
- `tctl apps ls`: `app_server`
- `tctl db ls`: `db_server`
- `tctl kube ls`: `kube_server`
- `tctl desktops ls --format text`: `windows_desktop` + `windows_desktop_service`
- `tctl desktops ls --format json|yaml`: `windows_desktop`  (with this PR)

How should we address this inconsistency in `tctl desktops ls`?

Reply:
I'm fine just removing the version of `tctl desktops ls --format=text`.

Technically, the desktops don't have a teleport version because they aren't running Teleport. If people want to see the version of `windows_desktop_service` they can use `tctl get windows_desktop_service` instead.

Reply:
Done in 6bfe41eed0d281429e221c0177ee1d42f9f5d88f. I updated the [top-level comment](https://github.com/gravitational/teleport/pull/18779#issue-1463812613) to reflect this change.